src/Components/RoutesWrapper.js

```diff
@@ -0,0 +1,45 @@
+import { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useNavigate, Routes, Route } from 'react-router-dom';
+import '../Style/myReservation.css';
+import { getExperiencesData } from '../Redux/Slices/ExperiencesSlice';
+import { getUserReservations } from '../Redux/Slices/reservationsSlice';
+import MainPage from './MainPage';
+import ReservationForm from './ReservationForm';
+import MyReservations from './MyReservations';
+import ExperienceCard from './ExperienceCard';
+
+function RoutesWrapper() {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const status = useSelector((state) => state.experiences.status);
+  const userStatus = useSelector((state) => state.users.status);
+  const { id } = useSelector((state) => state.users.user);
+
+  useEffect(() => {
+    if (status === 'idle') {
+      dispatch(getExperiencesData());
+    }
+  }, [dispatch, status]);
+
+  useEffect(() => {
+    if (status === 'succeeded') {
+      dispatch(getUserReservations(id));
+    }
+  });
+
+  if (userStatus === 'Authenticated') {
+    navigate('/login');
+  }
```

Comment:
I can notice all your effort and good work in this feature, but when you navigate to the `MainPage` I get this error that is related to the multiple uses of `useSelector`.
![image](https://github.com/HFG43/final_capstone_front_end/assets/119966922/677bdced-3905-4d0b-ab64-92d1edfeb949)

To solve this problem consider supplying the following lines of code

```ruby
const userStatus = useSelector((state) => state.users.status);
const { id } = useSelector((state) => state.users.user);

# Suplay the two previous lines with the following one that will give you access to all the states of `users`
const userStore = useSelector((state) => state.users);
``` 

```ruby
dispatch(getUserReservations(id));
if (userStatus === 'Authenticated') 

# Look up for those lines and replace with
dispatch(getUserReservations(userStore.user.id));
if (userStore.status === 'Authenticated') 
```